dotCMS/src/main/java/com/dotcms/security/secret/ServiceIntegrationAPIImpl.java

```diff
@@ -0,0 +1,554 @@
+                    final Host host = hostAPI.find(hostId, user, false);
+                    deleteSecrets(serviceKey, host, user);
+                }
+            }
+            if(removeDescriptor) {
+                final String fileName = serviceDescriptorMeta.getFileName();
+                //Now we need to remove the file it self.
+                final String ymlFilesPath = getServiceDescriptorDirectory();
+                final Path file = Paths.get(ymlFilesPath + File.separator + fileName).normalize();
+                if (!file.toFile().exists()) {
+                    throw new DotDataException(
+                            String.format(" File with path `%s` does not exist. ", file));
+                }
+                try {
+                    Logger.debug(ServiceIntegrationAPIImpl.class, () -> String
+                            .format(" Failed attempt to delete file with path `%s` ", file));
+                    Files.delete(file);
+                } catch (IOException e) {
+                    throw new DotDataException(e);
+                }
+            }
+            invalidateCache();
+        }
+    }
+
+    private synchronized void invalidateCache(){
+        CacheLocator
+                .getCacheAdministrator()
+                .remove(DESCRIPTORS_LIST_KEY, DESCRIPTORS_CACHE_GROUP);
+        CacheLocator.getCacheAdministrator()
+                .remove(DESCRIPTORS_MAPPED_BY_SERVICE_KEY, DESCRIPTORS_CACHE_GROUP);
+    }
+
+    private static String getServiceDescriptorDirectory() {
+        final Supplier<String> supplier = () -> APILocator.getFileAssetAPI().getRealAssetsRootPath()
+                + File.separator + "services" + File.separator;
+        final String dirPath = Config
+                .getStringProperty(SERVICE_INTEGRATION_DIR_PATH_KEY, supplier.get());
+        return Paths.get(dirPath).normalize().toString();
+    }
+
+    private Set<String> listAvailableYamlFiles() throws IOException {
+        final String ymlFilesPath = getServiceDescriptorDirectory();
+        final File basePath = new File(ymlFilesPath);
+        if (!basePath.exists()) {
+            basePath.mkdir();
+        }
+        Logger.debug(ServiceIntegrationAPIImpl.class,
+                () -> " ymlFiles are set under:  " + ymlFilesPath);
+        final Set<String> files = listFiles(ymlFilesPath);
+        if (!UtilMethods.isSet(files)) {
+            // No files were found a default empty descriptor will be created.
+            // ymlMapper.writeValue(new File(basePath, "sample-service-descriptor.yml"), emptyDescriptor());
+            // return listFiles(ymlFilesPath);
+           return Collections.emptySet();
+        } else {
+            return files;
+        }
+    }
+
+    private Set<String> listFiles(final String dir) throws IOException {
+        final Set<String> fileList = new HashSet<>();
+        try (final DirectoryStream<Path> stream = Files.newDirectoryStream(Paths.get(dir))) {
```

Comment:
I think the newDirectoryStream already receives a filter so you can avoid the for and filtering, something such as
```
Files.newDirectoryStream(Paths.get(dir), path ->
    Files.isRegularFile(path) &&  (path.toString().endsWith("yaml") || fileName.endsWith("yml")
).map(Path::toString)..collect(Collectors.toSet());
```